### PR TITLE
Made test timeout to be configurable by setting KNEX_TEST_TIMEOUT env…

### PR DESCRIPTION
**Pull request by elhigu**
…ironment variable.

### CODE REVIEW

**Comment by elhigu** [comment]
✓ accepts the table name, and a "container" function (7075ms) 👍

File: test/index.js [hunk 1]
```diff
@@ -18,6 +18,8 @@ global.d      = new Date();
 Promise.longStackTraces();
 
 describe('Query Building Tests', function() {
+  this.timeout(process.env.KNEX_TEST_TIMEOUT || 5000);
```
Comment (jurko-gospodnetic) [inline test/index.js]: I have not tested this, but would we be able to kill the duplication by making this call in a global `beforeEach()` call? As I recall, mocha has a top-level implicit anonymous test suite containing everything.

Comment (elhigu) [inline test/index.js]: I had no idea about that implicit top level. I'll try it out 👍

Comment (elhigu) [inline test/index.js]: Adding beforeEach() outside of describes was ran before every test, but setting timeout there didn't work:

```js
/*global describe*/

'use strict';

require('source-map-support').install();

global.sinon = require("sinon");

var chai = global.chai = require("chai");

chai.use(require("sinon-chai"));
chai.should();

var Promise   = global.testPromise = require('../lib/promise');
global.expect = chai.expect;
global.d      = new Date();

Promise.longStackTraces();

beforeEach(function () {
  console.log("This log is ran before every test, but setting timeout here has no effect!");
  this.timeout(process.env.KNEX_TEST_TIMEOUT || 5000);
})

describe('Query Building Tests', function() {
  require('./unit/query/builder')
  require('./unit/schema/mysql')('mysql')
  require('./unit/schema/mysql')('maria')
  require('./unit/schema/mysql')('mysql2')
  require('./unit/schema/postgres')
  require('./unit/schema/sqlite3')
  require('./unit/schema/oracle')
  require('./unit/schema/mssql')
})

describe('Integration Tests', function() {
  require('./integration')
})
```

Comment (jurko-gospodnetic) [inline test/index.js]: Doh... ah well... thanks for trying. :+1:

File: index.html [hunk 2]
```diff
@@ -3096,6 +3096,15 @@ <h2 id="faq">F.A.Q.</h2>
       replacing with the path to your config file, and the config file is valid, the test suite should run properly.
     </p>
 
+      <b class="header">My tests are failing because slow DB connection and short test timeouts! How to extend test timeouts?</b><br />
+      Some times e.g. with running CI on travis test suites default timeout 5seconds might be too short. The test suite looks from an environment variable 
+      called <tt>KNEX_TEST_TIMEOUT</tt> for milliseconds how long each test might run before timing out.
```
Comment (jurko-gospodnetic) [inline index.html]: Suggested doc text update:

> Sometimes, e.g. when running CI on travis, test suite's default timeout of 5 seconds might be too short. In such cases an alternative test timeout value in milliseconds can be specified using the <tt>KNEX_TEST_TIMEOUT</tt> environment variable.

(hmmm, I already wrote up this comment before but it's not here now... must not have confirmed it or something...)

Comment (elhigu) [inline index.html]: done, also fixed one missing paragraph starter tag

**Comment by jurko-gospodnetic** [comment]
:+1:

File: test/index.js (hunk 1 above)
Comment (wubzz) [inline test/index.js]: According to [Mocha Docs](http://mochajs.org/#timeouts) calling `.timeout(ms)` in a describe will apply that timeout for all nested tests in that suite. So maybe it'll work by adding `.timeout()` under `describe('Query Building Tests')` / `describe('Integration Tests')` ?

I know it's already merged, just figured I'd mention. :smile:

**Pull request merged**